pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -57,20 +59,21 @@ const (
 // ReconcileLaunchTemplate reconciles a launch template and triggers instance refresh conditionally, depending on
 // changes.
 //
-//nolint:gocyclo
+//nolint:gocyclo,maintidx
 func (s *Service) ReconcileLaunchTemplate(
```

Comment:
The PR contains quite some interface changes, parameter additions, etc. So I'm marking a few of the most important spots where the actual feature is implemented.

Main change number 1